okapi-core/src/test/java/org/folio/okapi/PullTest.java

```diff
@@ -50,17 +52,18 @@ private void setupOtherHttpServer(TestContext context, Async async) {
     Router router = Router.router(vertx);
 
     HttpServerOptions so = new HttpServerOptions().setHandle100ContinueAutomatically(true);
-    vertx.createHttpServer(so)
-      .requestHandler(router)
-      .listen(
-        port3,
-        result -> {
-          if (result.failed()) {
-            context.fail(result.cause());
-          }
-          async.complete();
-        }
-      );
+
+    otherServer = vertx.createHttpServer(so)
+        .requestHandler(router)
+        .listen(
+            port3,
+            result -> {
+              if (result.failed()) {
+                context.fail(result.cause());
+              }
+              async.complete();
+            }
+        );
```

Comment:
Use futurisation to avoid all the manual error handling:

    @Before
    public void setUp(TestContext context) {
      logger.debug("staring PullTest");
      vertx = Vertx.vertx();

      DeploymentOptions opt1 = new DeploymentOptions()
          .setConfig(new JsonObject().put("port", Integer.toString(port2)));
      DeploymentOptions opt2 = new DeploymentOptions()
          .setConfig(new JsonObject().put("port", Integer.toString(port1)));
      HttpServerOptions so = new HttpServerOptions().setHandle100ContinueAutomatically(true);

      vertx.deployVerticle(MainVerticle.class, opt1)
          .compose(x -> vertx.deployVerticle(MainVerticle.class, opt2))
          .compose(x -> vertx.createHttpServer(so)
              .requestHandler(Router.router(vertx))
              .listen(port3))
          .onComplete(context.asyncAssertSuccess());
    }